### PR TITLE
Feature/matrix1d form

### PR DESCRIPTION
**Pull request by tnagorra**

### CODE REVIEW

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix1dWidgetForm/index.tsx [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+    fields: (): RowSchemaFields => ({
+        clientId: [],
+        label: [requiredStringCondition],
+        tooltip: [],
+        color: [],
+        cells: cellsSchema,
+    }),
+    validation: (row) => {
+        if ((row?.cells?.length ?? 0) <= 0) {
+            return 'At least one cell is required.';
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetEditor/Matrix1dWidgetForm/index.tsx:89]: Use translation here

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix1dWidgetForm/index.tsx [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+                    name="title"
+                    value={value.title}
+                    onChange={onValueChange}
+                    error={error?.fields?.title}
+                />
+                <DataInput
+                    name="data"
+                    value={value.data}
+                    onChange={onValueChange}
+                    // eslint-disable-next-line max-len
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetEditor/Matrix1dWidgetForm/index.tsx]: remove this plz

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix1dWidgetForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+                <NonFieldError error={error?.fields?.rows} />
+                {value?.rows?.map((row, index) => (
+                    <RowInput
+                        className={styles.rowInput}
+                        key={row.clientId}
+                        index={index}
+                        value={row}
+                        onChange={onRowsChange}
+                        onRemove={onRowsRemove}
+                        // eslint-disable-next-line max-len
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetEditor/Matrix1dWidgetForm/index.tsx]: remove this plz

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix1dWidgetForm/index.tsx [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+                <NonFieldError error={error?.fields?.cells} />
+                {value.cells?.map((cell, cellIndex) => (
+                    <CellInput
+                        className={styles.cellInput}
+                        key={cell.clientId}
+                        index={cellIndex}
+                        value={cell}
+                        onChange={onCellsChange}
+                        onRemove={onCellsRemove}
+                        // eslint-disable-next-line max-len
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetEditor/Matrix1dWidgetForm/index.tsx]: remove this plz

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix1dWidgetForm/index.tsx [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+
+    const {
+        onValueChange: onRowsChange,
+        onValueRemove: onRowsRemove,
+    } = useFormArray('rows', onFieldChange);
+
+    const handleAdd = useCallback(
+        () => {
+            const oldRows = value?.rows ?? [];
+            // NOTE: Don't let users add more that certain statements
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetEditor/Matrix1dWidgetForm/index.tsx]: rows?

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx [hunk 6]
```diff
@@ -0,0 +1,806 @@
...
+    fields: (): DataSchemaFields => ({
+        rows: rowsSchema,
+        columns: columnsSchema,
+    }),
+    validation: (data) => {
+        if ((data?.rows?.length ?? 0) <= 0) {
+            return 'At least one row is required.';
+        }
+        if ((data?.columns?.length ?? 0) <= 0) {
+            return 'At least one column is required.';
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx:169]: Add FIXMEs for translation

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx [hunk 7]
```diff
@@ -0,0 +1,806 @@
...
+
+    const {
+        onValueChange: onSubRowsChange,
+        onValueRemove: onSubRowsRemove,
+    } = useFormArray('subRows', onFieldChange);
+
+    const handleAdd = useCallback(
+        () => {
+            const oldSubRows = value.subRows ?? [];
+            // NOTE: Don't let users add more that certain statements
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx]: sub rows?

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx [hunk 8]
```diff
@@ -0,0 +1,806 @@
...
+                <NonFieldError error={error?.fields?.subRows} />
+                {value.subRows?.map((subRow, subRowIndex) => (
+                    <SubRowInput
+                        className={styles.subRowInput}
+                        key={subRow.clientId}
+                        index={subRowIndex}
+                        value={subRow}
+                        onChange={onSubRowsChange}
+                        onRemove={onSubRowsRemove}
+                        // eslint-disable-next-line max-len
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx]: remove this

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx [hunk 9]
```diff
@@ -0,0 +1,806 @@
...
+
+    const {
+        onValueChange: onSubColumnsChange,
+        onValueRemove: onSubColumnsRemove,
+    } = useFormArray('subColumns', onFieldChange);
+
+    const handleAdd = useCallback(
+        () => {
+            const oldSubColumns = value.subColumns ?? [];
+            // NOTE: Don't let users add more that certain statements
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx]: subColumns?

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx [hunk 10]
```diff
@@ -0,0 +1,806 @@
...
+
+    const {
+        onValueChange: onColumnsChange,
+        onValueRemove: onColumnsRemove,
+    } = useFormArray('columns', onFieldChange);
+
+    const handleColumnAdd = useCallback(
+        () => {
+            const oldColumns = value?.columns ?? [];
+            // NOTE: Don't let users add more that certain statements
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx]: columns?

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx [hunk 11]
```diff
@@ -0,0 +1,806 @@
...
+
+    const {
+        onValueChange: onRowsChange,
+        onValueRemove: onRowsRemove,
+    } = useFormArray('rows', onFieldChange);
+
+    const handleRowAdd = useCallback(
+        () => {
+            const oldRows = value?.rows ?? [];
+            // NOTE: Don't let users add more that certain statements
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx]: rows?

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx [hunk 12]
```diff
@@ -0,0 +1,806 @@
...
+                <NonFieldError error={error?.fields?.columns} />
+                {value?.columns?.map((column, index) => (
+                    <ColumnInput
+                        className={styles.columnInput}
+                        key={column.clientId}
+                        index={index}
+                        value={column}
+                        onChange={onColumnsChange}
+                        onRemove={onColumnsRemove}
+                        // eslint-disable-next-line max-len
```
Comment (AdityaKhatri) [inline src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx]: remove this

**Review by AdityaKhatri — CHANGES_REQUESTED** [review]
Remove unnecessary lint disables and fix notes

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix1dWidgetForm/index.tsx (hunk 5 above)
Comment (tnagorra) [reply src/newViews/AnalyticalFramework/WidgetEditor/Matrix1dWidgetForm/index.tsx]: Oh yes!

File: src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx (hunk 7 above)
Comment (tnagorra) [reply src/newViews/AnalyticalFramework/WidgetEditor/Matrix2dWidgetForm/index.tsx]: that should have been items :V

**Pull request merged**